From fd73271a0a0a125d7aed71ea7aa013cc3f6c04b9 Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Wed, 25 Jan 2017 02:09:04 +0100
Subject: [PATCH 2/5] sepolicy: Allow recovery to alter /

This is needed for /etc/fstab, /adb_keys and volmgr

Change-Id: I53332a57ce7879d7ba63c4ea3e27add01f5a3a90
---
 prebuilts/api/33.0/public/domain.te | 4 ++--
 public/domain.te                    | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/prebuilts/api/33.0/public/domain.te b/prebuilts/api/33.0/public/domain.te
index 90e0592a2..40d1d6f0c 100644
--- a/prebuilts/api/33.0/public/domain.te
+++ b/prebuilts/api/33.0/public/domain.te
@@ -512,8 +512,8 @@ neverallow { domain -kernel with_asan(`-asan_extract') } { system_file_type vend
 # Don't allow mounting on top of /system files or directories
 neverallow * exec_type:dir_file_class_set mounton;
 
-# Nothing should be writing to files in the rootfs.
-neverallow * rootfs:file { create write setattr relabelto append unlink link rename };
+# Nothing should be writing to files in the rootfs, except recovery.
+neverallow { domain -recovery } rootfs:file { create write setattr relabelto append unlink link rename };
 
 # Restrict context mounts to specific types marked with
 # the contextmount_type attribute.
diff --git a/public/domain.te b/public/domain.te
index 90e0592a2..40d1d6f0c 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -512,8 +512,8 @@ neverallow { domain -kernel with_asan(`-asan_extract') } { system_file_type vend
 # Don't allow mounting on top of /system files or directories
 neverallow * exec_type:dir_file_class_set mounton;
 
-# Nothing should be writing to files in the rootfs.
-neverallow * rootfs:file { create write setattr relabelto append unlink link rename };
+# Nothing should be writing to files in the rootfs, except recovery.
+neverallow { domain -recovery } rootfs:file { create write setattr relabelto append unlink link rename };
 
 # Restrict context mounts to specific types marked with
 # the contextmount_type attribute.
-- 
2.30.2

